src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -1389,6 +1389,15 @@ public static Path copy(Path source, Path target, CopyOption... options)
      *     Files.move(source, newdir.resolve(source.getFileName()), REPLACE_EXISTING);
      * }
      *
+     * @apiNote
+     * If the options array contains the {@code REPLACE_EXISTING} option, but
+     * the file cannot be moved as an atomic file system operation, then it
+     * might be moved by first deleting the target file and then copying or
+     * renaming the source file to the target location. If some other process
+     * creates a file at the target location after the deletion but before the
+     * copying or renaming, then a {@code FileAlreadyExistsException} may be
+     * thrown despite the {@code REPLACE_EXISTING} option having been specified.
+     *
```

Comment:
I wonder if it might be better expand the description of REPLACE_EXISTING. That is, the existing paragraph could be expanded to say that if ATOMIC_MOVE is not specified then the checking the check for an existing file and the move is not atomic with respect to other filesystem activities.

Reply:
Done in 22d13f6c29aaafa56b51e0f03c4469d7d2d25f6b. I put the verbiage after the options table as REPLACE_EXISTING is ignored if ATOMIC_MOVE is specified so it did not seem accurate to put the text in the row describing the former option.